### DISCUSSION TITLE
Search by location / place not retrieveing all items

### DISCUSSION THREAD

**Discussion by cvinhaes** [photoprism/photoprism#2723] [Q&A] [answered]
I'm running the official docker image of Photoprism, version output is 220901-f493607b0-Linux-AMD64. Instance seems to be running properly, logs show indexing happening as expected. I have thousand of pictures taken in the area I live and all of them show the location properly in the card view, lat and long are present. However, when I search for the name of the location, only 15 pictures will display.

Most, if not all photos I reviewed have the Place attribute showing the location properly, however, it seems that search will only find the ones which the name of the place somehow ended in the title or keyword attributes. Is this the expected behavior? If yes, why place will show as a keyword for some photos and not others?

**Comment by graciousgrey** [comment]
Keywords are created from various sources, including, for example, the file and folder names or reverse geocoding. So if files have the same lat/lng but different file and folder names, the keywords may differ.

**Reply by cvinhaes** [reply]
This is not about the keywords, but the ability to search files by location. Metadata already knows the correct location of the files, nevertheless, search will only find these files if, for any reason, the location is present as a keyword. Search should be able to find the files based on their known location, that is my point. The files show correctly on the map but a search using the same name (place, city) won't. If I understand properly, country, state, lat and long can be used as a search filter, but city cannot, even when the city was properly retrieved by geo tagging.

**Reply by lastzero** [reply]
City and state have been added as search filters in our development preview, see Release Notes. These fields are not indexed and thus will not be searched, unless you explicitly use the corresponding filter 

However, the words found in the location name should be added to the fulltext index. Would have to invest some time to figure this out in detail together with you. If you can read the source, you should also be able to debug it on your own. Might be fastest as we don't have resources available at this time.

**Comment by cvinhaes** [comment]
Got it, thanks for the explanation.

For further clarity, here are the keywords for a photo that I cannot find with an unfiltered search, the name of the place in this example is 'Sao Conrado':
![image](https://user-images.githubusercontent.com/50474774/191739791-79c2ea03-81ac-4f16-9532-ec0e8e4f658d.png)

and I searched for 'conrado' with no success:
![image](https://user-images.githubusercontent.com/50474774/191740507-419be4e3-958c-456a-affe-887525ad5ba1.png)

A search for 'sao-conrado' will work but for 'conrado' or "sao conrado" won't. In this case, the match seems to be keyword-based only.

**Reply by lastzero** [reply, marked as answer]
That's possible as we implemented a simple text search that works on top of all databases including Sqlite. If I would have implemented this for a commercial client, I would have used Solr for search which is much more powerful. For example, it could easily do a phonetic comparison as well, so compare terms by how they sound like. In the past, city names like New York were split into two keywords. So now there is a dash as a workaround as that's what we came up with together with the user who reported the problem. Maybe there's another quick win to improve it if you check the source?
